extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -315,21 +318,17 @@ class QueueProcessor extends EventEmitter {
             const replicationStorageClass =
                 sourceEntry.getReplicationStorageClass();
             const sites = replicationStorageClass.split(',');
-            return async.each(sites, (site, cb) => {
+
+            if (sites.includes(this.site)) {
                 const replicationEndpoint = this.destConfig.bootstrapList
-                    .find(endpoint => endpoint.site === site);
-                if (replicationEndpoint
-                    && replicationBackends.includes(replicationEndpoint.type)) {
-                    task = new MultipleBackendTask(this, site);
+                    .find(endpoint => endpoint.site === this.site);
+                if (replicationEndpoint &&
+                replicationBackends.includes(replicationEndpoint.type)) {
+                    task = new MultipleBackendTask(this);
```

Comment:
No need to adapt `MultipleBackendTask` constructor? Because I don't see the change in this PR.

Reply:
The constructor change should be reflected in `ReplicateObject` and site is passed through the `getStateVars`

Reply:
Right, there is no explicit constructor in `MultipleBackendTask`

Reply:
In that case, doesn't the `getStateVars` method have to be updated to send the site?

Reply:
yup, passed as var name `site` from `getStateVars`